### DISCUSSION TITLE
Problem with publishing my mini-app in Farsi/Persian language!

### DISCUSSION THREAD

**Discussion by amirrouhandeh** [zepp-health/zeppos-forum#275] [Developer Feedback]
Hello there,

I have developed a mini-app that displays the current date in Iran's calendar in Farsi/Persian language. However, since there is no "fa-IR" language option available, I have set the default language of my app to English. 
![Screenshot 2023-12-05 110337](https://github.com/zepp-health/zeppos-forum/assets/132583970/1a4f4c50-4d11-463d-b6ff-57df16e7088f) 

The problem is that when I submit my app and select Iran as the target country, the default language is automatically set to English, which leads to my app being rejected due to the lack of an English language option. 

![Screenshot 2023-12-05 110330](https://github.com/zepp-health/zeppos-forum/assets/132583970/daf086d0-710c-4035-aefc-0371dba58f53)

I am seeking your assistance in resolving this issue so that I can publish my app successfully. Thank you.

**Comment by MyoungXUE** [comment]
Hello, it is difficult to upload the application because Persian is not supported.

I came up with this solution.

1. You can indicate in the application description that the application only supports Persian
2. Because Zepp's reviewers have difficulty understanding Persian, it may be difficult to verify the accuracy of this program. You need to leave your contact information, such as email, in the application description so that users can give you feedback when they encounter problems
3. I will communicate with colleagues in charge of application review to ensure that your Mini Program will not be rejected again due to language issues

**Reply by amirrouhandeh** [reply]
sounds great. I have just resubmitted it for revision. My app ID is 1031833, in case you need to refer to it. Thank you for your help.

**Comment by amirrouhandeh** [comment]
Thanks for your support, my app was published yesterday but unfortunately, I can't find it on the app store, is there any direct link available for the app?

**Reply by MyoungXUE** [reply]
@amirrouhandeh Hello, I checked that this application is only available in Iran. Can you provide your Zepp ID? I can check your account region.

**Reply by amirrouhandeh** [reply]
Sure, here is my Zepp ID : 5003269938
Is it possible to make the app available globally or will it be rejected due to its language?

**Reply by MyoungXUE** [reply]
Mainly because Zepp OS currently does not support Persian. If you switch to the global region, it will be rejected during the review due to language reasons.

**Reply by MyoungXUE** [reply]
I checked and your account region is not in Iran, which makes it impossible to view it in the app store.

**Reply by amirrouhandeh** [reply]
What criteria is involved to specify the region? IP? Phone Location? Build Number?

**Reply by MyoungXUE** [reply]
When registering an account in the Zepp App, the region will be selected, and the region of the account will be determined at that time and can not be modified later.

My suggestion is that you can register a new account in Iran for testing.

**Reply by amirrouhandeh** [reply]
It's perfect, thanks for your suggestions and support✌🏻

**Comment by amirrouhandeh** [comment]
Hey there,

I hope you're doing well.

Since I launched this app a few months ago, I've received numerous emails from users who own the amazfit-t-rex-2 and are encountering an issue with my app. The problem they're experiencing is that when they try to add the app shortcut on their watch, their devices freeze and then reboot.

I have tested the app on the simulator and it works without any issues. I'm reaching out to you to help identify any potential bugs that the OS or the app might have.

Thank you in advance for your attention.

**Reply by MyoungXUE** [reply]
we will investigate this issue. Thank you for your support.

**Reply by amirrouhandeh** [reply]
Any progress?
Since I still receive numerous emails issuing this problem.
